sql/src/main/java/io/crate/planner/consumer/CrossJoinConsumer.java

```diff
@@ -0,0 +1,415 @@
+                        jobId,
+                        ImmutableList.<Projection>of(),
+                        orderBy,
+                        previousOutputs,
+                        orderBy.orderBySymbols(),
+                        leftDQL,
+                        context.plannerContext()
+                );
+            } else {
+                mergePhase = PlanNodeBuilder.localMerge(
+                        jobId,
+                        ImmutableList.<Projection>of(),
+                        leftDQL,
+                        context.plannerContext());
+            }
+            mergePhase.executionNodes(localExecutionNodes);
+            return mergePhase;
+        }
+
+        private void sortQueriedTables(final Map<Object, Integer> relationOrder, List<QueriedTable> queriedTables) {
+            Collections.sort(queriedTables, new Comparator<QueriedTable>() {
+                @Override
+                public int compare(QueriedTable o1, QueriedTable o2) {
+                    return Integer.compare(
+                            MoreObjects.firstNonNull(relationOrder.get(o1.tableRelation()), Integer.MAX_VALUE),
+                            MoreObjects.firstNonNull(relationOrder.get(o2.tableRelation()), Integer.MAX_VALUE));
+                }
+            });
+        }
+
+        private QueriedTable getQueriedTable(MultiSourceSelect statement,
+                                             QualifiedName relationName,
+                                             TableRelation tableRelation) {
+            QueriedTable queriedTable = QueriedTable.newSubRelation(relationName, tableRelation, statement.querySpec());
+            queriedTable.normalize(analysisMetaData);
+            return queriedTable;
+        }
+
+        /**
+         * returns a map with the relation as keys and the values are their order in occurrence in the order by clause.
+         *
+         * e.g. select * from t1, t2, t3 order by t2.x, t3.y
+         *
+         * returns: {
+         *     t2: 0                 (first)
+         *     t3: 1                 (second)
+         * }
+         */
+        private Map<Object, Integer> getRelationOrder(MultiSourceSelect statement) {
+            OrderBy orderBy = statement.querySpec().orderBy();
+            if (orderBy == null || !orderBy.isSorted()) {
+                return Collections.emptyMap();
+            }
+
+            final Map<Object, Integer> orderByOrder = new IdentityHashMap<>();
+            int idx = 0;
+            for (Symbol orderBySymbol : orderBy.orderBySymbols()) {
+                for (AnalyzedRelation analyzedRelation : statement.sources().values()) {
+                    QuerySplitter.RelationCount relationCount = QuerySplitter.getRelationCount(analyzedRelation, orderBySymbol);
+                    if (relationCount != null && relationCount.numOther == 0 && relationCount.numThis > 0) {
+                        orderByOrder.put(analyzedRelation, idx);
```

Comment:
imho: select * from t1, t2, t3 order by t2.x, t3.y, t2.z

returns: {
   t3: 1
   t2: 2
} which is wrong